rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -424,6 +424,31 @@ rosbag2_storage::BagMetadata MCAPStorage::get_metadata()
     metadata_.topics_with_message_count.push_back(topic_info);
   }
 
+  const auto & mcap_metadatas = mcap_reader_->metadataIndexes();
+  auto range = mcap_metadatas.equal_range("rosbag2");
+  mcap::Status status;
+  mcap::Record mcap_record;
```

Comment:
```suggestion
  mcap::Record mcap_record{};
```
Nitpick. For the sake of the defensive programming.

Reply:
I'm adding this - _but_ I want to register my rebuttal that I don't know of a recommendation to always use an empty-brace initializer for default initialization. There is nothing recommending this in the Google C++ style guide, and there is no ambiguity in the syntax. Maybe you are thinking of preferring `Foo bar{}` to the "most vexing parse" `Foo bar()`. However, syntactically `Foo bar;` is equal to `Foo bar{};`, they both are unambiguous ways to call the default constructor.

Reply:
Ok, here the things why I am asking to put `{}` for this particular variable.
The `mcap::record` is defined as 
```
struct MCAP_PUBLIC Record {
  OpCode opcode;
  uint64_t dataSize;
  std::byte* data;

  uint64_t recordSize() const {
    return sizeof(opcode) + sizeof(dataSize) + dataSize;
  }
};
```
The statement that `"syntactically Foo bar; is equal to Foo bar{};"` is incorrect in this particular case. Because this is a structure POD type without a default constructor. 

If we will write just `mcap::Record mcap_record` usually on `arm` platforms in release mode the value of the 
`mcap_record.dataSize` and `mcap_record.data` will be un-initialized and will contain garbage.  
Compilers don't have obligations to init this POD at all in this case. e.g. Some compilers versions in some cases still do this default initialization but this fully up to the compiler.  
If define `mcap_record` as `mcap::Record mcap_record{};` with `{}` - the compiler will be forced to call ether a default constructor or explicitly initialize each member of the POD style with default value i.e `"0"`.

Reply:
Also, clang-tidy inspection highlighting this particular variable
![image](https://github.com/ros2/rosbag2/assets/7294722/39c8e467-7c52-4541-911c-b13736f18874)

Reply:
Interesting subtle point - thanks for the info